keras_hub/src/models/image_text_detector.py

```diff
@@ -0,0 +1,277 @@
+import keras
+import numpy as np
+
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.models.task import Task
+
+
+@keras_hub_export("keras_hub.models.ImageTextDetector")
+class ImageTextDetector(Task):
+    """Base class for all text detection tasks.
+
+    `ImageTextDetector` tasks wrap a `keras_hub.models.Task` and
+    a `keras_hub.models.Preprocessor` to create a model that can be used for
+    image segmentation.
+
+    All `ImageTextDetector` tasks include a `from_preset()` constructor which
+    can be used to load a pre-trained config and weights.
+
+    Args:
+        detection_thresh: The value for thresholding predicted mask outputs.
+            Defaults to 0.3.
+        min_area: Minimum area for a polygon to be considered valid. Defaults
+            to 10.0.
+        unclip_ratio: Expansion ratio of for the detected polygons.
+            Defaults to 3.0.
+    """
+
+    def __init__(
+        self, detection_thresh=0.3, min_area=10.0, unclip_ratio=2.0, **kwargs
+    ):
+        self.detection_thresh = detection_thresh
+        self.min_area = min_area
+        self.unclip_ratio = unclip_ratio
+        super().__init__(**kwargs)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "detection_thresh": self.detection_thresh,
+                "min_area": self.min_area,
+                "unclip_ratio": self.unclip_ratio,
+            }
+        )
+        return config
+
+    def postprocess_to_polygons(self, masks, contour_finder="simple"):
```

Comment:
We might want to consider rolling this into the preprocessing layer. That is what we do for generation.